### ISSUE TITLE
Regression in Link Layer Control Procedure (LLCP)

### ISSUE THREAD

**Issue by caco3** [zephyrproject-rtos/zephyr#50587]
As requested in https://discord.com/channels/720317445772017664/733036956392357908/1022831217969733712 I create following bug report.

We have an nrf52840 system where we use BLE to transfer data at a high througput.
After we updated from https://github.com/zephyrproject-rtos/zephyr/commit/e1c2c36f569a8e06184a2c5574f1d25ba00df8d7 to the next commit https://github.com/zephyrproject-rtos/zephyr/commit/e1c2c36f569a8e06184a2c5574f1d25ba00df8d7, we see following log messages after connecting (device is peripheral):
```
<wrn> bt_hci_core: opcode 0x2016 status 0x0c
err> bt_hci_core: Failed read remote features (-5)
```

Beside of those 2 log entries we did not see any negative impact!

As suggested by @cvinayak, with `CONFIG_BT_LL_SW_LLCP_LEGACY` enabled, the log entries are gone again.

**To Reproduce**
1. have an application with BLE
2. connect to it, eg using the [nRF Connect for Desktop](https://www.nordicsemi.com/Products/Development-tools/nrf-connect-for-desktop)
3. After connecting to the device, the mentioned log entries are shown in the device's log.

**Impact**
None as far as I can see

*** Log ***
Directly after connecting (only the last 2 lines are from Zephyr):
```
[00:00:30.326,171] <dbg> bt: connected: Connected via connection (0) at ED:ED:F1:F5:F0:9F (random)
[00:00:30.326,293] <dbg> bt: setTxPower: Actual TX Power: 8 dBm
[00:00:30.326,416] <dbg> bt: connected: Connection (0) - TX Power = 8 dBm
[00:00:30.326,416] <dbg> bt: connected: Update 2MBit PHY
[00:00:30.326,568] <inf> bt: Setting the Connection Parameters: Interval min: 7ms, Interval max: 7ms, Latency: 0, timout: 100ms...
[00:00:30.326,568] <inf> bt: Stopping Advertisement...
[00:00:30.326,599] <inf> bt: Advertising successfully stopped
[00:00:30.326,812] <wrn> bt_hci_core: opcode 0x2016 status 0x0c
[00:00:30.326,812] <err> bt_hci_core: Failed read remote features (-5)

```

**Environment (please complete the following information):**
Target: nrf52840
OS: Linux (Ubuntu)
Toolchain: Zephyr SDK 0.15.0
Commit SHA or Version used: v3.1.0 (https://github.com/zephyrproject-rtos/zephyr/commit/e1c2c36f569a8e06184a2c5574f1d25ba00df8d7)

**Comment by kruithofa** [comment]
I haven't been  able to reproduce the behaviour yet, but I think the problem may be solved by increasing the number of available local procedure contexts, for example by adding the following lines to your prj.conf file:

> CONFIG_BT_CTLR_ADVANCED_FEATURES=y
> CONFIG_BT_CTLR_LLCP_LOCAL_PROC_CTX_BUF_NUM=8

Or by running 'west build -t menuconfig':

- under 'sub Systems and OS Services -> Bluetooth' enable 'show advanced features', 
- under 'sub Systems and OS Services -> Bluetooth -> Advanced features' increase the number of local controle procedure contexts available to 8 (default = 4)

@caco3 is this something you can try?

**Comment by caco3** [comment]
Dear @kruithofa 

Thank you for looking into it.

I have  `CONFIG_BT_CTLR_ADVANCED_FEATURES` already activated because we need it for `CONFIG_BT_CTLR_ULL_HIGH_PRIO` and `CONFIG_BT_CTLR_XTAL_THRESHOLD`.

I now disabled `CONFIG_BT_LL_SW_LLCP_LEGACY` again and instead set `CONFIG_BT_CTLR_LLCP_LOCAL_PROC_CTX_BUF_NUM=8`.

The log entries then no longer show up and I didn't find any regression.

**Comment by kruithofa** [comment]
Thanks! I suggest that for the time being you keep this setting (i.e. CONFIG_BT_CTLR_LLCP_LOCAL_PROC_CTX_BUF_NUM=8). 
I'll leave the issue open until I've discussed/thought more about whether we need a better fix for this (or if increasing the number of buffers is sufficient)

**Comment by kruithofa** [comment]
@kruithofa increase the default to 6 (from 4)

**Comment by caco3** [comment]
I will test it in 2 weeks after my holidays

**Comment by caco3** [comment]
I can confirm that `CONFIG_BT_CTLR_LLCP_LOCAL_PROC_CTX_BUF_NUM=6` also is fine.

Note that in my case (on `3.2.0`), the default value is on `2`. 
Setting it to `4` also worked.

**Comment by cvinayak** [comment]
Thank you for confirming.

**Comment by caco3** [comment]
IMO we can close this ticket now.

**Comment by kruithofa** [comment]
Issue will be closed when PR #51300 is merged
